dwave/system/composites/embedding.py

```diff
@@ -515,6 +516,186 @@ def sample(self, bqm, chain_strength=1.0, chain_break_method=None,
+            Where S and T are edgelists or NetworkX Graphs.
+
+       embedding_parameters (dict, optional, default={}):
+            Parameter dictionary to be passed to the embedding method.
+
+    """
+    def __init__(self, child_sampler, embedding_method=minorminer,
+                 **embedding_parameters):
+        if not isinstance(child_sampler, dimod.Structured):
+            raise dimod.InvalidComposition("ParamEmbeddingComposite should "
+                                           "only be applied to a "
+                                           "Structured sampler")
+        self._children = [child_sampler]
+        self._embedding_method = embedding_method
+        self._embedding_parameters = embedding_parameters
+
+        self.embedding = None
+        self.child_response = None
+
+    @property
+    def children(self):
+        """list [child_sampler]: List containing the structured sampler."""
+        return self._children
+
+    @property
+    def parameters(self):
+        """dict[str, list]: Parameters in the form of a dict.
+
+        #TODO: Ideally, the `embedding_method` object should also have a
+        `parameters` attribute.
+        """
+
+        param = self.child.parameters.copy()
+        param['force_embed'] = []
+        param['chain_strength'] = []
+        param['chain_break_method'] =  []
+        param['chain_break_fraction'] = []
+        # Ideally
+        # param['embedding_parameters'] = self._embedding_method.parameters.copy()
+
+        return param
+
+    @property
+    def properties(self):
+        """dict: Properties in the form of a dict.
+        """
+        return {'child_properties': self.child.properties.copy()}
+
+    def get_embedding(self, bqm, target_edgelist=None,
+                      force_embed=False,
+                      embedding_method=None,
+                      **embedding_parameters):
+        """Retrieve or create a minor-embedding from BinaryQuadraticModel
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            target_edgelist (list, optional, default=<Child Structure>):
+                An iterable of label pairs representing the edges in the target graph.
+
+            force_embed (bool, optional, default=False):
+                If the sampler has an embedding return it. Otherwise, embed problem.
+
+            **embedding_parameters:
+                Parameters for the embedding method.
+
+        Returns:
+            embedding (dict):
+                Dictionary that maps labels in S_edgelist to lists of labels in the
+                graph of the structured sampler.
+        """
+        if not isinstance(bqm, dimod.BinaryQuadraticModel):
+            raise ValueError("get_embedding() only takes "
+                                   "dimod.BinaryQuadraticModel as input")
+
+        embedding = self.embedding
```

Comment:
I think that keeping the last embedding around is a useful feature, but I think that it makes more sense to have a more general caching embedding composite and might be out of scope for this composite.

Reply:
Makes sense, and I see that it's done with the Lazy/Fixed Embedding methods.
I guess the workflow I had in mind was if I wanted to "renew" the embedding at some point. What would be the suggested way to do this? I'm thinking...
```
structsampler = ...
sampler = LazyFixedEmbeddingComposite(structsampler)
for ... :
    response = sampler.sample(..., , embedding_parameters={...})
    ...
...
sampler.embedding = None
for ... :
    response = sampler.sample(..., embedding_parameters={...})
    ...
...
```

Another way would be to use FixedEmbeddingComposite and explicitly using:
```
embedding = find_embedding( ... ) 
sampler.embedding = embedding
```
But this removes the nice abstraction given by the composite.